DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepositoryND.cs

```diff
@@ -228,6 +228,64 @@ public DeviceRegistryRepositoryND(IConfigurationProvider configProvider, IDocDbR
             return device;
         }
 
+        /// <summary>
+        /// Updates an existing device in the DocumentDB
+        /// Throws a DeviceNotRegisteredException is the device does not already exist in the DocumentDB
+        /// </summary>
+        /// <param name="device"></param>
+        /// <returns></returns>
+        public async Task<DeviceND> UpdateDeviceAsyncND(DeviceND device)
+        {
+            if (device.id == null)
+            {
+                throw new DeviceRequiredPropertyNotFoundException("'DeviceID' property is missing");
+            }
+
+            string deviceId = device.id;
+
+            DeviceND existingDevice = await GetDeviceAsyncND(deviceId);
+
+            if (existingDevice == null)
+            {
+                throw new DeviceNotRegisteredException(deviceId);
+            }
+
+            string incomingRid = device._rid ?? "";
+
+            if (string.IsNullOrWhiteSpace(incomingRid))
+            {
+                // copy the existing _rid onto the incoming data if needed
+                var existingRid = existingDevice._rid ?? "";
+                if (string.IsNullOrWhiteSpace(existingRid))
+                {
+                    throw new InvalidOperationException("Could not find _rid property on existing device");
+                }
+                device._rid = existingRid;
+            }
+
+            string incomingId = deviceId ?? "";
```

Comment:
This is still not fixed. Please fix this. DeviceID and incomingId are different. See the original implementation.